src/ShipEngine.php

```diff
@@ -94,96 +115,101 @@ public function validateAddresses(array $addresses)
      * @link https://docs.shipengine.com/docs/list-your-carriers
      *
      * @return Carriers\Carrier[]
-     * @throws Exception\ApiErrorResponse
-     * @throws Exception\ApiRequestFailed
+     * @throws RequestException
+     * @throws ClientException
      */
     public function listCarriers()
     {
-        $response = $this->requestFactory->listCarriers()->send();
+        $response = $this->client->get('carriers');
 
         return array_map(function ($carrier) {
             return new Carriers\Carrier($carrier);
-        }, $response->getData('carriers'));
+        }, json_decode($response->getBody()->getContents(), 1)['carriers']);
     }
 
     /**
      * Get a single Carrier
      *
      * @param string $carrierId
      * @return Carriers\Carrier
-     * @throws Exception\ApiErrorResponse
-     * @throws Exception\ApiRequestFailed
+     * @throws RequestException
+     * @throws ClientException
      */
     public function getCarrier(string $carrierId)
     {
-        $response = $this->requestFactory->getCarrier($carrierId)->send();
+        $endpoint = sprintf("carriers/%s", $carrierId);
+        $response = $this->client->get($endpoint);
 
-        return new Carriers\Carrier($response->getData());
+        return new Carriers\Carrier(json_decode($response->getBody()->getContents(), 1));
     }
 
     /**
      * @param CarrierCode $carrierCode
      * @param string $carrierId
      * @return bool
-     * @throws Exception\ApiErrorResponse
-     * @throws Exception\ApiRequestFailed
+     * @throws RequestException
+     * @throws ClientException
      */
     public function removeCarrier(CarrierCode $carrierCode, string $carrierId): bool
     {
-        $response = $this->requestFactory->deleteCarrier($carrierCode, $carrierId)->send();
+        $endpoint = sprintf('connections/carriers/%1$s/%2$s', $carrierCode, $carrierId);
+        $response = $this->client->delete($endpoint);
 
-        return $response->isSuccessful();
+        return $response->getStatusCode();
```

Comment:
Need to change to a bool response.

Reply:
This actually made me relook at the response class, killed it off. Getting bool now. The response class was made for the cURL responses. That has changed.